packages/repo-tools/src/commands/api-reports/api-reports.ts

```diff
@@ -45,7 +45,15 @@ export const buildApiReports = async (paths: string[] = [], opts: Options) => {
   const runTsc = opts.tsc;
   const allowWarnings = parseArrayOption(opts.allowWarnings);
   const allowAllWarnings = opts.allowAllWarnings;
-  const omitMessages = parseArrayOption(opts.omitMessages);
+  const omitMessagesRaw = opts.omitMessages;
+
+  if (omitMessagesRaw && omitMessagesRaw.endsWith(',')) {
```

Comment:
Hm.

I mean the input was truly malformed; having a space there (in _any_ command line parsing system) means it was two separate arguments. If you'd passed in `-o "a, b"` with quotes, then I'd just say that `parseArrayOption` should instead be amended with a `.filter(Boolean)` to strip out accidental extra commas etc so that it's nicely forgiving. But if there's an unquoted space, the command correctly treats it as a folder argument to make reports for.

And THERE is actually the real bug.

```shell
yarn build:api-reports:only directoryThatDoesNotExist packages/cli
```

will happily generate reports for `packages/cli` but NOT complain about the first bad argument. If you could make a fix instead for that problem, it'd solve your problem as well as another really confusing thing that's been around forever. 🙏

Reply:
👍, I think it'd be reasonable to throw if a provided directory does not exist